Kernel/Arch/aarch64/Processor.cpp

```diff
@@ -118,6 +120,28 @@ void Processor::flush_tlb(Memory::PageDirectory const*, VirtualAddress vaddr, si
     flush_tlb_local(vaddr, page_count);
 }
 
+void Processor::leave_critical()
+{
+    InterruptDisabler disabler;
+    current().do_leave_critical();
```

Comment:
why not to inline the method? not sure about this `current()`—thinking about the situation where you have a handle to a different processor…? 🤔

noticed that it is also copied but i must say i still do not comprehend it…! 😔

Reply:
Right, I just copied the code from the x86_64 Processor.cpp. Might not be completely semantical correct, as you noticed, but can be improved later.